var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -96,20 +83,20 @@ class Dealii(CMakePackage):
 
     # optional dependencies
     depends_on("mpi",              when="+mpi")
-    depends_on("arpack-ng+mpi",    when='+arpack+mpi')
+    depends_on("arpack-ng",        when='+arpack')
     depends_on("doxygen+graphviz", when='+doc')
     depends_on("graphviz",         when='+doc')
     depends_on("gsl",              when='@8.5.0:+gsl')
-    depends_on("hdf5+mpi",         when='+hdf5+mpi')
+    depends_on("hdf5",             when='+hdf5')
     depends_on("metis@5:",         when='+metis')
-    depends_on("netcdf+mpi",       when="+netcdf+mpi")
-    depends_on("netcdf-cxx",       when='+netcdf+mpi')
+    depends_on("netcdf",           when="+netcdf")
+    depends_on("netcdf-cxx",       when='+netcdf')
     depends_on("oce",              when='+oce')
-    depends_on("p4est",            when='+p4est+mpi')
```

Comment:
I did separate things into 3 different commits for this reason. It sounds like the first 2 commits are popular, but the 3rd is much more controversial.